src/script/builder/tir/block_frame.h

```diff
@@ -62,15 +64,56 @@ class BlockFrameNode : public TIRFrameNode {
 };
 
 class BlockFrame : public TIRFrame {
+ public:
+  BlockFrame() = default;
+
  public:
   TVM_DEFINE_MUTABLE_NOTNULLABLE_OBJECT_REF_METHODS(BlockFrame, TIRFrame, BlockFrameNode);
 };
 
-BlockFrame Block_(String name);
+BlockFrame Block_(String name, bool no_realize = false);
+
+class BlockInitFrameNode : public TIRFrameNode {
+ public:
+  Optional<tvm::tir::Stmt> init;
+
+  void VisitAttrs(tvm::AttrVisitor* v) {
+    TIRFrameNode::VisitAttrs(v);
+    v->Visit("init", &init);
+  }
+
+  static constexpr const char* _type_key = "script.builder.tir.BlockInitFrame";
+  TVM_DECLARE_FINAL_OBJECT_INFO(BlockInitFrameNode, TIRFrameNode);
+
+ public:
+  void EnterWithScope() final;
+  void ExitWithScope() final;
+};
+
+class BlockInitFrame : public TIRFrame {
+ public:
+  TVM_DEFINE_MUTABLE_NOTNULLABLE_OBJECT_REF_METHODS(BlockInitFrame, TIRFrame, BlockInitFrameNode);
+};
+
+BlockInitFrame BlockInit();
+
+tvm::tir::BufferRegion BufferRegionFromLoad(tvm::tir::BufferLoad buffer_load);
+void BlockWhere(PrimExpr predicate);
+void BlockReads(Array<ObjectRef> buffer_slices);
+void BlockWrites(Array<ObjectRef> buffer_slices);
+void BlockAttrs(Map<String, ObjectRef> attrs);
```

Comment:
To be consistent with TVMScript parser, which uses:

```python
T.where
T.reads
T.writes
```

Let's rename to:

```suggestion
void Where(PrimExpr predicate);
void Reads(Array<ObjectRef> buffer_slices);
void Writes(Array<ObjectRef> buffer_slices);
void BlockAttrs(Map<String, ObjectRef> attrs);
```